File: Part 2 - Regression/Section 9 - Random Forest Regression/Random_Forest_Regression/Random_Forest_Regression/my_random_forest_regression.R
# Random Forest Regression

dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Fitting Decision Tree Regression to the dataset
#install.packages('randomForest')
library(randomForest)
set.seed(1234)
regressor = randomForest( x = dataset[1], y = dataset$Salary, ntree = 500)



# Predicting a new result
y_pred = predict(regressor, data.frame(Level = 6.5))
cat(sprintf("X: 6.5, y_pred: %f\n", y_pred))


# Visualising the Decision Tree Regression results
library(ggplot2)
X_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
p <- ggplot() + geom_point(aes(x = dataset$Level, y = dataset$Salary),colour = 'red')
p <- p	+ geom_line(aes(x = X_grid, y = predict(regressor, newdata = data.frame(Level = X_grid))),colour = 'blue')
p <- p	+ ggtitle('Salary vs Experience(Random Forest Regression)')
p <- p	+ xlab('Level') 
p <- p	+ ylab('Salary')
plot(p)

File: Part 2 - Regression/Section 9 - Random Forest Regression/Random_Forest_Regression/Random_Forest_Regression/my_random_forest_regression.py
#!/usr/bin/env python3
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
Y = dataset.iloc[:, 2].values
X_grid = np.arange(min(X), max(X), 0.01)
X_grid = X_grid.reshape(-1, 1)


from sklearn.ensemble import RandomForestRegressor
n_layers = 0
n_layers_gap = 500
n_plot = 4
for i in range (1,1 + n_plot):
	n_layers = i * n_layers_gap
	regressor2 = RandomForestRegressor(n_estimators = n_layers, random_state = 0)
	regressor2.fit(X, Y)
	
	y_pred2 = regressor2.predict(6.5)
	print("X: 6.5, y_pred: %f(n_estimators = %d)"%(y_pred2, n_layers))
	
	plt.subplot(1, n_plot, i)
	plt.scatter(X, Y, color = 'red')
	plt.plot(X_grid, regressor2.predict(X_grid), color = 'blue')
	str_title = "Truth or Bluff(Random Forest n_layer %d)" % n_layers
	plt.title(str_title)
	plt.xlabel('Position Level')
	plt.ylabel('Salary')


plt.show()

